docs/root/configuration/other_features/reverse_tunnel.rst

```diff
@@ -7,170 +7,116 @@ Reverse Tunnels
-and keep them alive for reuse. These connections are established using a handshake protocol and can be
-used for forwarding traffic from services behind upstream Envoy to downstream services behind the downstream Envoy.
+Envoy supports reverse tunnels that enable establishing persistent connections from downstream Envoy
+instances to upstream Envoy instances without requiring the upstream to be directly reachable from the
+downstream. This feature is particularly useful in scenarios where downstream instances are behind NATs,
+firewalls, or in private networks, and need to communicate with upstream instances in public networks
+or cloud environments.
+
+Reverse tunnels invert the typical connection model: the downstream Envoy initiates TCP connections to
+upstream Envoy instances and keeps them alive for reuse. These connections are established using a
+handshake protocol, after which traffic can be forwarded bidirectionally. Services behind the upstream
+Envoy can send requests through the tunnel to downstream services behind the initiator Envoy, effectively
+treating the normally unreachable downstream services as if they were directly accessible.
+
+.. image:: images/reverse_tunnel_arch.svg
+   :width: 90%
+   :align: center
 
 .. _config_reverse_tunnel_bootstrap:
 
 Reverse tunnels require the following extensions:
 
-1. **Downstream socket interface**: Registered as a bootstrap extension on initiator Envoy to initiate and maintain reverse tunnels.
-2. **Upstream socket interface**: Registered as a bootstrap extension on responder Envoy to accept and manage reverse tunnels.
-3. **Reverse tunnel network filter**: On responder Envoy to accept reverse tunnel requests.
-4. **Reverse connection cluster**: Added on responder Envoy for each downstream Envoy node that needs to be reached through reverse tunnels.
+1. **Downstream socket interface**: Registered as a bootstrap extension on the initiator Envoy to initiate and maintain reverse tunnels.
+2. **Upstream socket interface**: Registered as a bootstrap extension on the responder Envoy to accept and manage reverse tunnels.
+3. **Reverse tunnel network filter**: Configured on the responder Envoy to accept and validate reverse tunnel handshake requests.
+4. **Reverse connection cluster**: Configured on the responder Envoy to route data requests to downstream nodes through established reverse tunnels.
 
 .. _config_reverse_tunnel_configuration_files:
 
-Configuration Files
--------------------
-
-For practical examples and working configurations, see:
-
-* :repo:`Initiator Envoy configuration <configs/reverse_connection/initiator-envoy.yaml>`: Configuration for the initiator Envoy (downstream)
-* :repo:`Responder Envoy configuration <configs/reverse_connection/responder-envoy.yaml>`: Configuration for the responder Envoy (upstream)
-
 .. _config_reverse_tunnel_initiator:
 
-Initiator Configuration (Downstream Envoy)
+Initiator configuration (Downstream Envoy)
```

Comment:
```suggestion
Initiator configuration (downstream Envoy)
```

i think this probably should just be lower but for sure it should be consistent with other usage